### PR TITLE
Update to the builder documentation

### PR DESCRIPTION
**Pull request by acozine**
This is far from final, but since lots of folks are talking about docs for `ansible-builder`, I thought I should put what I have so far out in the public domain.

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
-  installed as well as the ``ansible-builder`` Python package.
-  The ``--container-runtime`` option needs to correspond to the Podman/Docker
-  executable you intend to use.
-- ``ansible-builder`` version ``1.1.0`` requires Python ``3.8`` or higher.
-
+- To build images, you must install either ``podman`` or ``docker``
+  as well as the ``ansible-builder`` Python package.
+- The ``--container-runtime`` option needs to correspond to the Podman/Docker
+  executable you intend to use (where is this set?).
+- ``ansible-builder`` version ``3.x`` requires Python ``???`` or higher.
```
Comment (Akasurde) [inline docs/installation.rst]:
```suggestion
- ``ansible-builder`` version ``3.x`` requires Python ``3.8`` or higher.
```

Comment (acozine) [reply docs/installation.rst]: wow, 1.1 and 3.x have the same Python version requirement?

File: docs/collection_metadata.rst [hunk 2]
```diff
@@ -1,66 +1,46 @@
-Collection-level Metadata
-=========================
+Collection-level Dependencies
```
Comment (samccann) [inline docs/collection_metadata.rst]:
```suggestion
Collection-level dependencies
```

File: docs/collection_metadata.rst [hunk 3]
```diff
@@ -1,66 +1,46 @@
...
+=============================
 
-Collections inside of the ``galaxy`` entry of an execution environment
-will contribute their Python and system requirements to the image.
+When Ansible Builder installs Collections into an Execution Environment, it also installs the dependencies listed by each Collection on Galaxy.
```
Comment (samccann) [inline docs/collection_metadata.rst]:
```suggestion
When Ansible Builder installs Collections into an execution environment, it also installs the dependencies listed by each Collection on Galaxy.
```

Comment (samccann) [reply docs/collection_metadata.rst]: I looked at the RH stuff and they use lower case for EEs when it's spelled out.

File: docs/collection_metadata.rst [hunk 4]
```diff
@@ -1,66 +1,46 @@
...
-Entries from separate collections that give the same *package name* will
-be combined into the same entry, with the constraints combined.
+If multiple collections require the same *package name*, Ansible Builder combines them into a single entry and combines the constraints.
 
-There are several package names which are specifically *ignored* by
-``ansible-builder``, meaning that if a collection lists these, they will
-not be included in the combined file. These include test packages and
-packages that provide Ansible itself. The full list can be found in
-``EXCLUDE_REQUIREMENTS`` in the ``ansible_builder.requirements`` module.
+Certain package names are specifically *ignored* by ``ansible-builder``, meaning that Ansible Builder does not include them in the combined file of Python dependencies, even if a collection lists them as dependencies. These include test packages and packages that provide Ansible itself. The full list can be found in ``EXCLUDE_REQUIREMENTS`` in the ``ansible_builder.requirements`` module.
```
Comment (samccann) [inline docs/collection_metadata.rst]: Is there a link or something for the `ansible_builder.requirements` module or am I missing something obvious here? I tried searching for it but can't find it...

File: docs/collection_metadata.rst [hunk 5]
```diff
@@ -1,66 +1,46 @@
...
-excluded Python packages.
+If you need to include one of these ignored package names, use the ``--user-pip`` option of the ``introspect`` command to list it in the user requirements file. Packages supplied this way are not processed against the list of excluded Python packages.
 
 System-level Dependencies
 ^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The ``bindep`` format provides a way of specifying cross-platform
-requirements. A minimum expectation is that collections specify
-necessary requirements for ``[platform:rpm]``.
+For system packages, use the ``bindep`` format to specify cross-platform requirements, so they can be installed by whichever package management system the Execution Environment uses. Collections should specify necessary requirements for ``[platform:rpm]``.
```
Comment (samccann) [inline docs/collection_metadata.rst]:
```suggestion
For system packages, use the ``bindep`` format to specify cross-platform requirements, so they can be installed by whichever package management system the execution environment uses. Collections should specify necessary requirements for ``[platform:rpm]``.
```

File: docs/collection_metadata.rst [hunk 6]
```diff
@@ -1,66 +1,46 @@
...
 
-Entries from multiple collections will be combined into a single file.
-Only requirements with *no* profiles (runtime requirements) will be
-installed to the image. Entries from multiple collections which are
-outright duplicates of each other may be consolidated in the combined
-file.
+Ansible builder combines system package entries from multiple collections into a single file. Only requirements with *no* profiles (runtime requirements) are installed to the image. Entries from multiple collections which are outright duplicates of each other may be consolidated in the combined file.
```
Comment (samccann) [inline docs/collection_metadata.rst]:
```suggestion
Ansible Builder combines system package entries from multiple collections into a single file. Only requirements with *no* profiles (runtime requirements) are installed to the image. Entries from multiple collections which are outright duplicates of each other may be consolidated in the combined file.
```

File: docs/definition.rst [hunk 7]
```diff
@@ -1,27 +1,94 @@
 Execution Environment Definition
 ================================
 
-The execution environment (EE) definition file supports multiple versions.
+You define the content of your Execution Environment in a YAML file. By default, this file is called ``execution_environment.yml``. This file tells Ansible Builder how to create the Containerfile and build your container image. The Execution Environment definition file accepts four top-level sections: version, images, dependencies, and (optionally) additional_build_steps.
```
Comment (samccann) [inline docs/definition.rst]:
```suggestion
You define the content of your execution environment in a YAML file. By default, this file is called ``execution_environment.yml``. This file tells Ansible Builder how to create the Containerfile and build your container image. The execution environment definition file accepts four top-level sections: version, images, dependencies, and (optionally) additional_build_steps.
```

File: docs/definition.rst [hunk 8]
```diff
@@ -1,27 +1,94 @@
...
+Example:
+
+.. code-block:: yaml
+
+   version: 3
 
 .. note::
+   If you are running an older version of Ansible Builder, you may need an older schema version. Please consult older versions of the docs for more information. We recommend using version 3, which offers substantially more configurability and functionality than previous versions.
+
+If the EE file does not specify a version, version 1 will be assumed. (IS THIS STILL TRUE???)
```
Comment (samccann) [inline docs/definition.rst]: @nitzmahone  can you confirm ^^ ?

File: docs/definition.rst [hunk 9]
```diff
@@ -1,27 +1,94 @@
...
+     system:
+       - python3-dnf
+
+
+additional_build_steps
+----------------------
+Custom build steps.
+Optional.
+
+You can add build steps before or after any stage of the image creation process. Each step should be expressed as a containerfile directive. For example, if you need ``git`` to be installed before you install your dependencies, you can add a build step at the end of the ``base`` build stage.
```
Comment (samccann) [inline docs/definition.rst]: I can't figure out if containerfile should be Containerfile here?

File: docs/index.rst [hunk 10]
```diff
@@ -5,55 +5,59 @@
 Introduction
 ============
 
-Using Ansible content that depends on non-default dependencies can be tricky.
-Packages must be installed on each node, play nicely with other software installed
-on the host system, and be kept in sync.
+With ``ansible-builder`` you can configure and build portable, consistent, customized Ansible `control nodes`_ that are packaged as containers by Podman or Docker. These containers are known as Execution Environments. You can use execution environments on AWX or Ansible Controller, for local playbook development and testing, in your CI pipeline, and anywhere else you run automation. You can design and distribute specialized Execution Environments for your playbooks, choosing the versions of Python and ansible-core you want, and installing only the python packages, system packages, and Ansible Collections you need for each playbook.
```
Comment (samccann) [inline docs/index.rst]:
```suggestion
With ``ansible-builder`` you can configure and build portable, consistent, customized Ansible `control nodes`_ that are packaged as containers by Podman or Docker. These containers are known as execution environments. You can use execution environments on AWX or Ansible Controller, for local playbook development and testing, in your CI pipeline, and anywhere else you run automation. You can design and distribute specialized execution environments for your playbooks, choosing the versions of Python and ansible-core you want, and installing only the Python packages, system packages, and Ansible Collections you need for each playbook.
```

File: docs/index.rst [hunk 11]
```diff
@@ -5,55 +5,59 @@
...
+Here are a few terms you should know. These concepts and terms are relevant to any use of containers, not specific to Ansible Builder.
+
+  * Container: a package of code and dependencies that runs a service or an application across a variety of computing environments
+  * Containerfile (called a Dockerfile in Docker): an instruction file for creating a container image by installing and configuring the code and dependencies  
+  * Image: a complete but inactive version of a container - you can distribute images and create one or more containers based on each image
 
 Execution Environments
 ----------------------
 
 Execution Environments are container images that serve as Ansible `control
```
Comment (samccann) [inline docs/index.rst]:
```suggestion
Execution environments are container images that serve as Ansible `control
```

File: docs/index.rst [hunk 12]
```diff
@@ -5,55 +5,59 @@
...
-nodes`_. Starting in version 2.0, `ansible-runner
-<https://ansible-runner.readthedocs.io/en/latest/execution_environments.html>`__
-can make use of these images.
+nodes`_. An Execution Environment contains:
```
Comment (samccann) [inline docs/index.rst]:
```suggestion
nodes`_. An execution environment contains:
```

File: docs/index.rst [hunk 13]
```diff
@@ -5,55 +5,59 @@
...
+- Python
+- Ansible-core
+- Ansible-runner
+- Ansible Collections, if needed
+- Python packages (anything you install with ``pip``)
+- System packages (anything you install with ``dnf``, ``yum``, ``apt``, or other package managers)
+- Other content you select and install
 
 .. _control nodes:
    https://docs.ansible.com/ansible/latest/network/getting_started/basic_concepts.html#control-node
```
Comment (samccann) [inline docs/index.rst]:
```suggestion
   https://docs.ansible.com/ansible/latest/getting_started/basic_concepts.html#control-node
```

File: docs/index.rst [hunk 14]
```diff
@@ -5,55 +5,59 @@
...
+execution environment. You can specify these items:
+- system packages (any packages installed with ``apt`` or ``yum``)
+- the version of Python
+- all Python packages (any packages installed with ``pip``)
+- the version of ansible-core
+- all Ansible Collections (anything installed with ``ansible-galaxy``)
+- other items to download, install, or configure
 
-Once you have created a definition, read through the :ref:`Usage:CLI Usage`.
+Ansible Builder reads the ``execution_environment.yml`` file and can execute two separate steps. The first step is to create a containerfile and a build context. The second step is to run a containerization tool (Podman or Docker) to build an image. You can use ``ansible-builder build`` to run both steps. Or you can use ``ansible-builder create`` to run only the first step. For more details, read through the :ref:`Usage:CLI Usage`.
```
Comment (samccann) [inline docs/index.rst]:
```suggestion
Ansible Builder reads the ``execution_environment.yml`` file and can execute two separate steps. The first step is to create a Containerfile and a build context. The second step is to run a containerization tool (Podman or Docker) to build an image. You can use ``ansible-builder build`` to run both steps. Or you can use ``ansible-builder create`` to run only the first step. For more details, read through the :ref:`Usage:CLI Usage`.
```

File: docs/index.rst [hunk 15]
```diff
@@ -5,55 +5,59 @@
...
 
 Collection Developers
 ^^^^^^^^^^^^^^^^^^^^^
 
-Collection developers can declare requirements for their content by providing
-the appropriate metadata. For more information, see
-:ref:`collection_metadata:Collection-level Metadata`.
+When Ansible Builder installs Collections into an Execution Environment, it also installs each Collection's dependencies. You can learn to correctly declare dependencies for your collection from the :ref:`collection_metadata:Collection-level Metadata` page.
```
Comment (samccann) [inline docs/index.rst]:
```suggestion
When Ansible Builder installs Collections into an execution environment, it also installs each Collection's dependencies. You can learn to correctly declare dependencies for your collection from the :ref:`collection_metadata:Collection-level Metadata` page.
```

Comment (samccann) [reply docs/index.rst]: what do you think about changing 'You can learn...' to "Collection owners can learn"  .. to separate the 'you' of a person building an EE from a person who created the collection(s)?

File: docs/installation.rst [hunk 16]
```diff
@@ -1,15 +1,17 @@
...
-- In order to build images, you need to have either ``podman`` or ``docker``
-  installed as well as the ``ansible-builder`` Python package.
-  The ``--container-runtime`` option needs to correspond to the Podman/Docker
-  executable you intend to use.
-- ``ansible-builder`` version ``1.1.0`` requires Python ``3.8`` or higher.
-
+- To build images, you must install either ``podman`` or ``docker``
+  as well as the ``ansible-builder`` Python package.
+- The ``--container-runtime`` option needs to correspond to the Podman/Docker
+  executable you intend to use (where is this set?).
```
Comment (samccann) [inline docs/installation.rst]: @Akasurde @nitzmahone can you answer this question ^^ ?

File: docs/usage.rst [hunk 17]
```diff
@@ -1,28 +1,25 @@
 CLI Usage
 =========
 
-Once you have created a :ref:`definition<Definition:Execution Environment Definition>`, it's time to build
-your Execution Environment.
+Ansible Builder does two things. First it creates a containerfile and a build context based on your :ref:`definition<Definition:Execution Environment Definition>`. Then it runs Podman or Docker to build an image based on the containerfile and build context. If you wish, you can use Ansible Builder only for the first task, then create your images on the platform of your choice.
```
Comment (samccann) [inline docs/usage.rst]:
```suggestion
Ansible Builder does two things. First it creates a Containerfile and a build context based on your :ref:`definition<Definition:Execution Environment Definition>`. Then it runs Podman or Docker to build an image based on the Containerfile and build context. If you wish, you can use Ansible Builder only for the first task, then create your images on the platform of your choice.
```

File: docs/usage.rst [hunk 18]
```diff
@@ -1,28 +1,25 @@
...
 
+The ``ansible-builder build`` command executes both steps, giving you a containerfile, a build context, and a fully built container image. The ``ansible-builder create`` command only executes the first step, giving you a containerfile and a build context.
```
Comment (samccann) [inline docs/usage.rst]:
```suggestion
The ``ansible-builder build`` command executes both steps, giving you a Containerfile, a build context, and a fully built container image. The ``ansible-builder create`` command only executes the first step, giving you a Containerfile and a build context. 
```

File: docs/usage.rst [hunk 19]
```diff
@@ -68,35 +66,28 @@ More recent versions of ``ansible-builder`` support multiple tags:
...
 
+If you do not pass this option, no signature verification is performed. If you do pass this option, but the version of Ansible is too old to support this feature, you will see an error during the image build process.
+
 ``--galaxy-ignore-signature-status-code``
 *****************************************
 
-With ``--galaxy-keyring`` set it is possible to ignore certain errors that may occur while verifying collections.
-It is passed unmodified to ``ansible-galaxy`` calls via the option ``--ignore-signature-status-code``.
-See the ``ansible-galaxy`` documentation for more information.
+When ``--galaxy-keyring`` is set, ignores certain errors that may occur while verifying collections. This option is passed unmodified to ``ansible-galaxy`` calls. See the ``ansible-galaxy`` documentation for more information.
```
Comment (samccann) [inline docs/usage.rst]:
```suggestion
When ``--galaxy-keyring`` is set,  Galaxy ignores certain errors that may occur while verifying collections. This option is passed unmodified to ``ansible-galaxy`` calls. See the ``ansible-galaxy`` documentation for more information.
```

File: docs/usage.rst [hunk 20]
```diff
@@ -106,9 +97,7 @@ See the ``ansible-galaxy`` documentation for more information.
 ``--galaxy-required-valid-signature-count``
 *******************************************
 
-When ``--galaxy-keyring`` is set, the number of required valid collection signatures can be overridden.
-The value is passed unmodified to ``ansible-galaxy`` calls via the option ``--required-valid-signature-count``.
-See the ``ansible-galaxy`` documentation for more information.
+When ``--galaxy-keyring`` is set, overrides the number of required valid collection signatures. This option is passed unmodified to ``ansible-galaxy`` calls. See the ``ansible-galaxy`` documentation for more information.
```
Comment (samccann) [inline docs/usage.rst]:
```suggestion
When ``--galaxy-keyring`` is set, Galaxy overrides the number of required valid collection signatures. This option is passed unmodified to ``ansible-galaxy`` calls. See the ``ansible-galaxy`` documentation for more information.
```

File: docs/usage.rst [hunk 21]
```diff
@@ -134,9 +122,9 @@ directory. To specify another location:
 ``--build-arg``
 ***************
 
-To use Podman or Docker's build-time variables, specify them the same way you would with ``podman build`` or ``docker build``.
+Passes build-time arguments to Podman or Docker. Specify these flags or variables the same way you would with ``podman build`` or ``docker build``.
 
-By default, the Containerfile / Dockerfile outputted by Ansible Builder contains a build argument ``EE_BASE_IMAGE``, which can be useful for rebuilding Execution Environments without modifying any files.
+By default, the Containerfile / Dockerfile created by Ansible Builder contains a build argument ``EE_BASE_IMAGE``, which can be useful for rebuilding Execution Environments without modifying any files.
```
Comment (samccann) [inline docs/usage.rst]:
```suggestion
By default, the Containerfile / Dockerfile created by Ansible Builder contains a build argument ``EE_BASE_IMAGE``, which can be useful for rebuilding execution environments without modifying any files.
```

File: docs/usage.rst [hunk 22]
```diff
@@ -209,22 +195,21 @@ To customize the level of verbosity:
...
 
 .. code::
 
    $ ansible-builder build --prune-images
 
 .. note::
 
-   This flag essentially removes all the dangling images on the given machine whether they
-   already exists or created by ansible-builder build process.
+   This flag removes all the dangling images on the given machine whether they already existed or were created by ansible-builder build process.
```
Comment (samccann) [inline docs/usage.rst]:
```suggestion
   This flag removes all the dangling images on the given machine whether they already existed or were created by ``ansible-builder`` build process.
```

**Review by samccann — COMMENTED** [review]
Thanks so much for this PR! Makes things a lot clearer to a novice like me.

I acted as capitalization police as I read through...ignore if you disagree etc

File: docs/collection_metadata.rst (hunk 4 above)
Comment (acozine) [reply docs/collection_metadata.rst]: Oh, nice catch! I just changed the formatting without changing the content here. I see the list in `src/ansible_builder/_target_scripts/introspect.py`. I'll update the text and add a link in my next changeset.

File: docs/definition.rst (hunk 9 above)
Comment (acozine) [reply docs/definition.rst]: Ah, yes, this probably needs to be fixed throughout. I think `containerfile` is the generic term and `Containerfile` and `Dockerfile` are the platform-specific terms.

File: docs/index.rst (hunk 15 above)
Comment (acozine) [reply docs/index.rst]: Good idea.

File: docs/installation.rst (hunk 16 above)
Comment (acozine) [reply docs/installation.rst]: I think I understand this now. I'll update the text in my next changeset.

File: docs/installation.rst (hunk 1 above)
Comment (acozine) [reply docs/installation.rst]: I'm going to assume @Akasurde is correct.

File: docs/usage.rst (hunk 19 above)
Comment (acozine) [reply docs/usage.rst]: Hm, this change isn't correct. The start of each flag description is `does this thing` - meaning, `setting this flag does this thing` - so it's your `ansible-builder` command that ignores the errors here, not Galaxy.

Comment (acozine) [reply docs/usage.rst]: But maybe I should move the `when x is set` phrase, to make this clearer.

File: docs/usage.rst (hunk 20 above)
Comment (acozine) [reply docs/usage.rst]: Ditto here.

File: docs/definition.rst (hunk 9 above)
Comment (acozine) [reply docs/definition.rst]: I think I've caught all of these. The new standard expression is "containerfile (Containerfile for Podman, Dockerfile for Docker)".

**Comment by acozine** [comment]
Thanks @Akasurde and @samccann for the suggestions so far! There's still some cleanup that could be done (those pesky instances of `containerfile` vs. `Containerfile`, etc.) and the page about the EE definition file needs a bunch more review/work, but this is starting to feel decent. I can't do any more today but could pick this up again next week.

@nitzmahone if you could do a first-round review, that would be great. I did my best to translate the technical details into plain English, but it would be good to find and fix any mistakes I made in the translation.

File: docs/index.rst [hunk 23]
```diff
@@ -5,55 +5,84 @@
...
+
+How Ansible Builder builds images
+---------------------------------
+
+Ansible Builder executes four stages when it runs your containerization tool to build a container image. The same four stages get executed if you build your container image directly with Podman or Docker, using a containerfile and context generated by ``ansible-builder create``. These stages are:
+- Base
+- Galaxy
+- Builder
+- Final
+In the Base stage, Ansible Builder uses Podman or Docker to pull the base image you defined. All three later stages of the build process build on the output of the Base stage. In the Galaxy stage, Ansible Builder downloads the collections you defined from Galaxy and stashes them locally as files. In the Builder stage, Ansible Builder downloads the other packages (python packages and system packages) you defined and stashes them locally as files. In the Final stage, Ansible Builder integrates the first three stages, installing all the stashed files on the base image and generating a new image that includes all the content.
```
Comment (samccann) [inline docs/index.rst]:
```suggestion
In the Base stage, Ansible Builder uses Podman or Docker to pull the base image you defined. All three later stages of the build process build on the output of the Base stage. In the Galaxy stage, Ansible Builder downloads the collections you defined from Galaxy and stashes them locally as files. In the Builder stage, Ansible Builder downloads the other packages (Python packages and system packages) you defined and stashes them locally as files. In the Final stage, Ansible Builder integrates the first three stages, installing all the stashed files on the base image and generating a new image that includes all the content.
```

File: docs/installation.rst (hunk 1 above)
Comment (Shrews) [reply docs/installation.rst]: 3.9 is actually the minimum for `ansible-builder 3.0`.

File: docs/definition.rst [hunk 24]
```diff
@@ -1,27 +1,94 @@
-Execution Environment Definition
+Execution environment definition
 ================================
 
-The execution environment (EE) definition file supports multiple versions.
+You define the content of your execution environment in a YAML file. By default, this file is called ``execution_environment.yml``. This file tells Ansible Builder how to create the containerfile (Containerfile for Podman, Dockerfile for Docker) and build context for your container image. The execution environment definition file accepts four top-level sections: version, images, dependencies, and (optionally) additional_build_steps.
```
Comment (Shrews) [inline docs/definition.rst]: I prefer something like `build instruction file` or even just `instruction file` over `containerfile`.

File: docs/definition.rst (hunk 8 above)
Comment (Shrews) [reply docs/definition.rst]: Yes, version 1 is assumed if missing. Not sure why that's in question.

File: docs/definition.rst (hunk 9 above)
Comment (Shrews) [reply docs/definition.rst]: Mentioned in another comment, but I prefer the term `instruction file` over `containerfile`. I pretty much rewrote this entire document for version 3 and used `instruction file` for the generic term.

File: docs/definition.rst [hunk 25]
```diff
@@ -1,27 +1,94 @@
...
-    documentation.
 
 Version 3 Format
 ----------------
 
-This version of the EE definition file offers substantially more configurability
-and functionality over previous versions.
-
-Below is an example version 3 EE file:
+This example file shows many more options from the version 3 EE file:
```
Comment (Shrews) [inline docs/definition.rst]: This doesn't read well to me (what example file?). I prefer the original text, tbh (seeing as I wrote it, I might be biased).

File: docs/installation.rst [hunk 26]
```diff
@@ -1,15 +1,17 @@
...
+- ``ansible-builder`` version ``3.x`` requires Python ``3.8`` or higher.
```
Comment (Shrews) [inline docs/installation.rst]: `3.9` is the minimum. See requirements in `setup.cfg`.

File: docs/definition.rst (hunk 25 above)
Comment (Shrews) [reply docs/definition.rst]: Oh, I suppose you mean the example below this. It still reads awkward/redundant ("file" shows more options from the "file" (i.e., itself)). Still prefer the original text.

File: docs/index.rst (hunk 23 above)
Comment (acozine) [reply docs/index.rst]: addressed in 1bbfeb03c2ff49e58e8c72fe64596078c4287d20 (I found a couple of other python/Python changes)

File: docs/installation.rst (hunk 26 above)
Comment (acozine) [reply docs/installation.rst]: Thanks for the review @Shrews! I think I have addressed your concerns and your answers in 4b5a9fd85759ecd7dcb4c4aad197de99a6e3b6f1 and 09bb9aff1880a7bc17822599191901d2ed584c4d.

**Comment by acozine** [comment]
I have taken this PR out of "draft" status. I believe I've addressed all comments so far.

File: docs/definition.rst (hunk 24 above)
Comment (acozine) [reply docs/definition.rst]: Changed, thanks for the feedback.

File: docs/definition.rst (hunk 25 above)
Comment (acozine) [reply docs/definition.rst]: Yep, this was awkward. Updated.

**Review by samccann — APPROVED** [review]
LGTM

File: docs/definition.rst [hunk 27]
```diff
@@ -1,27 +1,31 @@
...
+Overview
+--------
+The Ansible Builder 3.x execution environment definition file accepts seven top-level sections:
+* additional_build_files
+* additional_build_steps
+* build_arg_defaults
+* dependencies
+* images
+* options
+* version
```
Comment (Shrews) [inline docs/definition.rst:24]: This doesn't display correctly.

FYI, the `docs/readthedocs.org:ansible-builder` job provides a preview of what the resulting documentation will look like (click `Details` next to the job).

Comment (acozine) [reply docs/definition.rst:24]: ah, needs a blank line, probably

File: docs/index.rst [hunk 28]
```diff
@@ -2,58 +2,85 @@
...
+To get started with Ansible Builder, you must install the ``ansible-builder`` utility and a containerization tool. Once you have the tools you need, create an :ref:`execution environment definition <builder_ee_definition>` file. By default this file is called ``execution_environment.yml``. In this file you can specify the exact content you want to include in your
+execution environment. You can specify these items:
+- the base container image
+- the version of Python
+- the version of ansible-core
+- the version of ansible-runner
+- Ansible collections, with version restrictions
+- system packages, with version restrictions
+- Python packages, with version restrictions
+- other items to download, install, or configure
```
Comment (Shrews) [inline docs/index.rst:57]: List is not rendered properly in the final docs.

File: docs/index.rst [hunk 29]
```diff
@@ -2,58 +2,85 @@
...
-To get started with Ansible Builder, start by creating an
-:ref:`Execution Environment definition<Definition:Execution Environment Definition>`. This
-file allows you to specify which content you would like to include in your
-execution environment, such as collections, Python requirements, and
-system-level packages.
+Ansible Builder executes four stages when it runs your containerization tool to build a container image. The same four stages get executed if you build your container image directly with Podman or Docker, using a build instruction file and context generated by ``ansible-builder create``. These stages are:
+- Base
+- Galaxy
+- Builder
+- Final
```
Comment (Shrews) [inline docs/index.rst:77]: Another list not rendered properly. (Dang lists! lol)

Thanks so much for adding this, though. We really need these stages explained.

File: docs/index.rst [hunk 30]
```diff
@@ -2,58 +2,85 @@
...
+In the Base stage, Ansible Builder uses Podman or Docker to pull the base image you defined. All three later stages of the build process build on the output of the Base stage. In the Galaxy stage, Ansible Builder downloads the collections you defined from Galaxy and stashes them locally as files. In the Builder stage, Ansible Builder downloads the other packages (Python packages and system packages) you defined and stashes them locally as files. In the Final stage, Ansible Builder integrates the first three stages, installing all the stashed files on the base image and generating a new image that includes all the content.
```
Comment (Shrews) [inline docs/index.rst]: During the base stage, not only is the base image pulled, but it is also customized for all later stages. E.g., ansible and ansible-runner are installed (if defined), alternate Python version is installed (if defined), and we make sure pip is available. There are no "stashed files" on the base image, either. Those are only from Galaxy and Builder intermediate stages.

**Review by Shrews — CHANGES_REQUESTED** [review]
Thank you so much for improving our docs! Just a few minor things that I found.

File: docs/index.rst (hunk 30 above)
Comment (acozine) [reply docs/index.rst]: Ah, thanks for the details! I hope 579613fefc2e2fe8da699244ed05e22552b22742 addresses all these comments @Shrews.

**Review by Shrews — APPROVED** [review]

**Pull request merged**